src/components/steps/steps.tsx

```diff
@@ -0,0 +1,213 @@
+import React, { useEffect, useState } from 'react';
+import IonIcon from '../icons/icons';
+import {
+  CircleStyle,
+  LineStyle,
+  DetailsStyle,
+  StepsContainerStyle,
+  StepStyle,
+  StepDrawStyle,
+} from './styles';
+
+export interface StepProps {
+  steps: any[];
+  handleClick?: () => void;
+}
+
+export type StatusType = 'default' | 'selected' | 'checked' | 'error';
+
+export interface StepType {
+  label: string;
+  description?: string;
+  index?: number;
+  status?: StatusType;
+}
+
+export type LineType = 'before' | 'after';
+
+export type StepConfig = {
+  current?: number;
+  disabled?: boolean;
+  steps: StepType[];
+  clickable?: boolean;
+};
+
+export type StepCircle = {
+  status?: StatusType;
+  index?: number;
+  steps: StepType[];
+};
+
+export type StepLine = {
+  status?: StatusType;
+  index?: number;
+  steps: StepType[];
+  type: LineType;
+};
+
+export type StepDetails = {
+  label: string;
+  description?: string;
+  index?: number;
+};
+
+/* eslint-disable no-unused-vars */
+enum Status {
+  default = 'default',
+  selected = 'selected',
+  checked = 'checked',
+  error = 'error',
+}
+
+function LineDraw({ type, status, index, steps }: StepLine) {
+  const beforeStepIsChecked = (index: number = 0): boolean => {
+    return steps[index - 2] && steps[index - 2].status === Status.checked;
+  };
+
+  const afterStepIsChecked = (index: number = 0): boolean => {
+    if (status && status === Status.checked) return true;
+    return steps[index + 2] && steps[index + 2].status !== Status.default;
+  };
+
+  if (type === 'before') {
+    return (
+      <>
+        {index !== 1 && (
+          <LineStyle bolded={beforeStepIsChecked(index)}></LineStyle>
```

Comment:
Components that don't have children can be defined in a simple tag `<LineStyle />`